edgelet/iotedge/src/main.rs

```diff
@@ -323,6 +351,96 @@ fn run() -> Result<(), Error> {
             tokio_runtime.block_on(Logs::new(id, options, runtime()?).execute())
         }
         ("version", _) => tokio_runtime.block_on(Version::new().execute()),
+        ("support-bundle", Some(args)) => {
+            let location = args.value_of("location").unwrap_or_default();
+            let since = args
+                .value_of("since")
+                .map(|s| parse_since(s))
+                .transpose()?
+                .unwrap_or_default();
+            let options = LogOptions::new()
+                .with_follow(false)
+                .with_tail(LogTail::All)
+                .with_since(since);
+            tokio_runtime
+                .block_on(SupportBundle::new(options, location.to_owned(), runtime()?).execute())
+        }
         (command, _) => tokio_runtime.block_on(Unknown::new(command.to_string()).execute()),
     }
 }
+
+fn parse_since(since: &str) -> Result<i32, Error> {
+    if let Ok(datetime) = DateTime::parse_from_rfc3339(since) {
+        let temp: Result<i32, _> = datetime.timestamp().try_into();
+        Ok(temp.context(ErrorKind::BadSinceParameter)?)
+    } else if let Ok(epoch) = since.parse() {
+        Ok(epoch)
+    } else if let Ok(duration) = parse_duration::parse(since) {
+        let nano: Result<i64, _> = duration.as_nanos().try_into();
+        let nano = nano.context(ErrorKind::BadSinceParameter)?;
+
+        let temp: Result<i32, _> = (Local::now() - Duration::nanoseconds(nano))
+            .timestamp()
+            .try_into();
+        Ok(temp.context(ErrorKind::BadSinceParameter)?)
+    } else {
+        Err(Error::from(ErrorKind::BadSinceParameter))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn parse_rfc3339() {
+        assert_eq!(
+            parse_since("2019-09-27T16:00:00+00:00").unwrap(),
+            1_569_600_000
+        );
+    }
+
+    #[test]
+    fn parse_english() {
+        assert_near(
+            parse_since("1 hour").unwrap(),
+            (Local::now() - Duration::hours(1))
+                .timestamp()
+                .try_into()
+                .unwrap(),
+            10,
+        );
+
+        assert_near(
+            parse_since("1 hour 20 minutes").unwrap(),
+            (Local::now() - Duration::hours(1) - Duration::minutes(20))
+                .timestamp()
+                .try_into()
+                .unwrap(),
+            10,
+        );
+
+        assert_near(
+            parse_since("1 day").unwrap(),
+            (Local::now() - Duration::days(1))
+                .timestamp()
+                .try_into()
+                .unwrap(),
+            10,
+        );
+    }
+
+    #[test]
+    fn parse_unix() {
+        assert_eq!(parse_since("1569600000").unwrap(), 1_569_600_000);
+    }
+
+    #[test]
+    fn parse_default() {
+        assert!(parse_since("asdfasdf").is_err());
```

Comment:
`let _ = parse_since("asdfasdf").unwrap_err();` will give you a better panic message like `called unwrap_err on an Ok value <>` than `expected true, got false`.